test/gateways/accountHolderApiGateway.test.ts

```diff
@@ -0,0 +1,45 @@
+import axios, { AxiosResponse } from "axios";
+import { v4 } from "uuid";
+import { AccountHolderApiGateway } from "../../src/gateways/accountHolderApiGateway";
+import { IAccountHolderIdResponseBody } from "../../src/lib/accountHolder/accountHolderIdResponseBody";
+
+jest.mock("axios");
+
+describe("Account Holder API Gateway", () => {
+  let gateway: AccountHolderApiGateway;
+  const endpoint = "account-holder/auth-id";
+
+  beforeEach(() => {
+    gateway = new AccountHolderApiGateway();
+  });
+
+  describe("Getting an account holder", () => {
+    let authId;
+    let acountHolderId;
+    let token;
+
+    beforeEach(() => {
+      authId = v4();
+      acountHolderId = v4();
+      token = v4();
+    });
+
+    it("should return account holder id", async () => {
+      const expectedUrl = `${process.env.API_URL}/${endpoint}/${authId}`;
+      (axios as any).get.mockImplementation(() => {
+        return {
+          data: {
+            id: acountHolderId,
+          },
+        } as AxiosResponse<IAccountHolderIdResponseBody>;
+      });
+
+      const expected = await gateway.getAccountHolderId(authId, token);
+
+      expect(expected).toBe(acountHolderId);
+      expect((axios as any).get).toHaveBeenLastCalledWith(expectedUrl, {
+        headers: { Authorization: `Bearer ${token}` },
+      });
```

Comment:
Is this testing two different behaviours?  I'd be tempted to separate these into two tests, but maybe I'm splitting hairs.  Something like:

1.  `"should request an accountHolderId from the correct endpoint"` which includes the expectation at line 40.
2. `"should return the obtained accountHolderId from the API"` which includes the expectation at line 39.

At the moment I think the test name only refers to the first expectation?

Reply:
done :)